text/0000-const-trait-impls.md

```diff
@@ -0,0 +1,1045 @@
+- Feature Name: `const_trait_methods`
+- Start Date: 2024-12-13
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#67792](https://github.com/rust-lang/rust/issues/67792)
+
+# Summary
+[summary]: #summary
+
+Make trait methods callable in const contexts. This includes the following parts:
+
+* Allow marking `trait` declarations as const implementable.
+* Allow marking `trait` impls as `const`.
+* Allow marking trait bounds as `const` to make methods of them callable in const contexts.
+
+Fully contained example ([Playground of currently working example](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=2ab8d572c63bcf116b93c632705ddc1b)):
```

Comment:
If we mean to allow this with RPIT and APIT, we probably want to add an example of this here also.